### ISSUE TITLE
Provide a solution for control-plane load-balancing

### ISSUE THREAD

**Issue by maelk** [metal3-io/cluster-api-provider-baremetal#106]
This issue is related to https://github.com/metal3-io/cluster-api-provider-baremetal/issues/101. When provisioning a cluster, we need to set up a load-balancer for the control-plane, since we may not be able to rely on a cloud-provider load-balancer. With this issue, we will go through the different alternatives to set up an HA cluster on baremetal nodes without external load-balancer.

We will start with a design document opened to discussion.

**Comment by maelk** [comment]
Here is a first draft document with the scope and goals [WIP] : https://docs.google.com/document/d/1LLywou05ak7df1VYQns27k70s6NgA8I-fILuJPDJZfI

**Comment by russellb** [comment]
We've done something for this in OpenShift.  Roughly the idea is that we bring up a cluster with a self-hosted load balancer and allow configuration of an external load balancer post-install if desired.

The self hosted load balancer solution for the API includes keepalived and haproxy.  First we require a virtual IP (VIP) to be allocated for this purpose.  The VIP is managed by keepalived across the masters.  haproxy is also running on all of the masters.  API traffic goes to the host where the VIP currently resides, and is redirected to one of the active hosts by haproxy.

**Comment by russellb** [comment]
@celebdor FYI - in case you'd like to add some more input here

**Comment by celebdor** [comment]
That is a very good description of what we do, @russellb . @maelk Let me know if you want to discuss that in more detail.

**Comment by maelk** [comment]
@celebdor yes I would be very interested in discussing it, to know how you did it in details and see if it is applicable for us.
